lib/phoenix/live_dashboard/info/modal.ex

```diff
@@ -0,0 +1,19 @@
+defmodule Phoenix.LiveDashboard.Modal do
+  @type param() :: binary()
+  @type decoded_param() :: term()
+  @callback decode_params(param()) :: {:ok, decoded_param()} | :error
+  @callback title(param(), decoded_param()) :: binary()
+  @optional_callbacks title: 2
+
+  defmacro __using__(opts) do
+    quote location: :keep, bind_quoted: [opts: opts] do
+      use Phoenix.LiveDashboard.Web, :live_component
+      @behaviour Phoenix.LiveDashboard.Modal
+
+      @impl Phoenix.LiveDashboard.Modal
+      def title(param, _), do: param
```

Comment:
If it is optional, don't define it. Instead, use function_exported?(model, :title, 2) to check if it exists, and then fallback to  `param` if it does not. :)